### PR TITLE
feat(swagger): Provide URI version to operationIdFactory 

### PR DESCRIPTION
**Pull request by deongroenewald**
This is based on the work completed in #1949. Those changes were reverted due to test failures.
This change adapts those changes to also support neutral versions which resolves the test failures.

Adds a new optional parameter to operationIdFactory to allow providing the API version prefix added to the operation's path.

This allows the operationIdFactory to generate unique IDs in the case of the same controller method being used for multiple API versions.

Default behavior is unchanged - the operation IDs will be of the form controllerKey_methodKey, but where necessary this can be customised using the URI version prefix.

Closes nestjs#2537

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?
When using Nest's API versioning feature, and having a controller method cover multiple versions, `nestjs-swagger `generates invalid OpenAPI spec, due to duplicate operation IDs.

It generates duplicate operation IDs because only the controller and method names are available to `operationIdFactory - but in these cases the same controller and method is used multiple times.

Issue Number: 2537

## What is the new behavior?

When URI versioning is used, an additional parameter containing the version is provided to `operationIdFactory`. This allows developers to provide unique operation IDs in the case of a method being used in multiple versions.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: lib/interfaces/swagger-document-options.interface.ts [hunk 1]
```diff
@@ -1,3 +1,15 @@
+export type OperationIdFactory =
```
Comment (kamilmysliwiec) [inline lib/interfaces/swagger-document-options.interface.ts:1]: Why union here?

Comment (deongroenewald) [reply lib/interfaces/swagger-document-options.interface.ts]: Oh, not sure how that slipped in. Removed.

File: lib/interfaces/swagger-document-options.interface.ts [hunk 2]
```diff
@@ -1,3 +1,9 @@
+export type OperationIdFactory = (
+  controllerKey: string,
+  methodKey: string,
+  pathVersionKey?: string
```
Comment (kamilmysliwiec) [inline lib/interfaces/swagger-document-options.interface.ts:4]:
```suggestion
  version?: string
```

File: lib/swagger-explorer.ts [hunk 3]
```diff
@@ -275,6 +279,25 @@ export class SwaggerExplorer {
...
+      } else if (versionOrVersions !== VERSION_NEUTRAL) {
+        versions = [versionOrVersions];
+      }
+
+      const versionConfig = applicationConfig.getVersioning();
+      if (versionConfig?.type == VersioningType.URI) {
+        const prefix = this.routePathFactory.getVersionPrefix(versionConfig);
+        versions = versions.map((v) => `${prefix}${v}`);
+      }
+    }
```
Comment (kamilmysliwiec) [inline lib/swagger-explorer.ts:299]: should we move this logic to a dedicated method?

**Comment by kamilmysliwiec** [comment]
LGTM

**Pull request merged**